pkg/app/piped/cmd/piped/piped.go

```diff
@@ -638,17 +638,17 @@ func (p *piped) sendPipedMeta(ctx context.Context, client pipedservice.Client, c
 
 	var (
 		req = &pipedservice.ReportPipedMetaRequest{
-			Version:        version.Get().Version,
-			Config:         string(maskedCfg),
-			Repositories:   repos,
-			CloudProviders: make([]*model.Piped_CloudProvider, 0, len(cfg.CloudProviders)),
```

Comment:
Is it ok to remove this field in gRPC?
As far as I see only CloudProvider is being used in the implementation of that RPC.
https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/grpcapi/piped_api.go#L167-L194

Reply:
Nice catch, the current implementation pass from the test because I updated the ReportPipedMetaRequest previously (in [here](https://github.com/pipe-cd/pipecd/blob/master/pkg/app/server/service/pipedservice/service.proto#L193)). 

The API update for that is included by another PR
```
--- a/pkg/app/server/grpcapi/piped_api.go
+++ b/pkg/app/server/grpcapi/piped_api.go
@@ -69,7 +69,7 @@ type pipedApiDeploymentChainStore interface {
 
 type pipedApiPipedStore interface {
        Get(ctx context.Context, id string) (*model.Piped, error)
-       UpdateMetadata(ctx context.Context, id, version, config string, cps []*model.Piped_CloudProvider, repos []*model.ApplicationGitRepository, se *model.Piped_SecretEncryption, startedAt int64) error
+       UpdateMetadata(ctx context.Context, id, version, config string, pps []*model.Piped_PlatformProvider, repos []*model.ApplicationGitRepository, se *model.Piped_SecretEncryption, startedAt int64) error
 }
 
 type pipedApiEventStore interface {
@@ -176,7 +176,7 @@ func (a *PipedAPI) ReportPipedMeta(ctx context.Context, req *pipedservice.Report
                pipedID,
                req.Version,
                req.Config,
-               req.CloudProviders,
+               req.PlatformProviders,
                req.Repositories,
                req.SecretEncryption,
                now,
```
but you're right, maybe we should include the change for that in this PR as well 🤔

Reply:
> maybe we should include the change for that in this PR as well

Yes, that is better. Or the PR for that API change should be merged before this PR.

Reply:
Okay, lets me revise this. Thank you 🙏

Reply:
addressed by https://github.com/pipe-cd/pipecd/pull/3815/commits/0272ba2daf59e1841f67a15755924cbf6b217388